README.md

```diff
@@ -64,3 +64,87 @@ part of a test, say if you wanted to check that the title number is displayed on
 a page then simply call the particular attribute for the deed you created:
 
 `page.should have_content(@deed.title_number)`
+
+
+## Calling steps from within steps  [AF 1/3/16]
+
+In Cucumber it is possible to call an existing step, from inside a different step body.
+These `helper steps` can be used to make feature files more readable, and to utilize
+existing code.
+
+example:
+`Given(/^I setup a deed with <(\d+)> borrowers$/) do |borrower|
+  # Creates a deed, with entered number of [borrowers].`
+  step %(I have valid deed data with <#{borrower}> borrowers)
+`end`
+
+Inside this step, the `I have valid deed data with <#{borrower}> borrowers` step is
+being called. It is using the parameter [borrower] being passed in by the parent step
+i.e `I setup a deed with <(\d+)> borrowers` to execute.
+
+
+## Amending a deed [AF 1/3/16]
+
+`Sample Deed`
+@borrowers=
+  [{:forename=>"Jayne",
+    :surname=>"Cobb",
+    :gender=>"Female",
+    :address=>"0B Borrower Street, Plymouth, PL3 2PP",
+    :dob=>"01/03/2016",
+    :phone_number=>"07063286046"}],
+ @identity_checked="Y",
+ @md_ref="e-MD12344",
+ @property_address="1 High Street, High Wycombe, CR0 6TY",
+ @title_number="Q87">
+
+New generic steps have been created to amend any of the above variables.
+The `testdebt_generic_test_data_step.feature` contains an example.
+
+For example if we want to amend the first name for a borrower on a deed:
+We can use:
+
+Given I setup a deed with <1> borrowers
+And I amend "forename" to "Timmy"
+And I create the deed
+
+OR
+
+Given I setup a deed with <1> borrowers
+And I amend "forename" to "Timmy" for borrower <1>
+And I create the deed
+
+Both are valid, because the step uses `Optional Values` see below.
+
+
+## Optional Values
+
+Cucumber supports the use of optional values, when writing steps.
+In the above example cucumber is expecting to see `And I amend "forename" to "Timmy"`
+but `for borrower <1>` is optional, so if it is present it knows how to process it, and
+likewise, if it is not present it can also process.
+
+This is occurs in the step name, by surrounding the optional value in (?:)?
+
+E.g `/^I amend "([^"]*)" to "([^"]*)"(?: for borrower <(\d+)>)?$/`
+
+Cucumber can understand this as
+1. /^I amend `"([^"]*)" to "([^"]*)?$/`  or
+2. /^I amend `"([^"]*)" to "([^"]*)" for borrower <(\d+)>)?$/`
+
+Depending on what is used in the feature file.
+
+
+## Default Values
+`Example`
+And(/^I retrieve the unique user id for borrower (?:<(\d+)>)?$/) do `|borrower|`
```

Comment:
Looking at the actual readme it seems like this line isn't quite formatted correctly with the markdown as I'm guessing you wanted the name of the step to be styled as code.